pmtiles/server.go

```diff
@@ -448,48 +485,85 @@ func parseMetadataPath(path string) (bool, string) {
 	return false, ""
 }
 
-// Get a response for the given path.
-// Return status code, HTTP headers, and body.
-func (server *Server) Get(ctx context.Context, path string) (int, map[string]string, []byte) {
-	httpHeaders := make(map[string]string)
+func (server *Server) get(ctx context.Context, path string) (archive, handler string, status int, headers map[string]string, data []byte) {
+	handler = ""
+	archive = ""
+	headers = make(map[string]string)
 	if len(server.cors) > 0 {
-		httpHeaders["Access-Control-Allow-Origin"] = server.cors
+		headers["Access-Control-Allow-Origin"] = server.cors
 	}
 
 	if ok, key, z, x, y, ext := parseTilePath(path); ok {
-		return server.getTile(ctx, httpHeaders, key, z, x, y, ext)
-	}
-	if ok, key := parseTilejsonPath(path); ok {
-		return server.getTileJSON(ctx, httpHeaders, key)
-	}
-	if ok, key := parseMetadataPath(path); ok {
-		return server.getMetadata(ctx, httpHeaders, key)
+		archive, handler = key, "tile"
+		status, headers, data = server.getTile(ctx, headers, key, z, x, y, ext)
```

Comment:
One concern here is that someone could make a bunch of random requests to nonexistent archives and all of the 404's for unique resources would blow up the metric response size. To avoid that, we could either:

1. try to map all "archive does not exist" responses to handler="404" archive="" (but we don't want to miss logging legit failures about a bad connection to an archive you care about)
2. or we could just recommend that applications put this behind a proxy that allows only requests to archives you know exist

Thoughts @bdon ?

Reply:
I would prefer 1). for 2) we would have to recommend they put the whole go-pmtiles application behind a allowlist-proxy, right? that is too strong of a limitation imho. 

for 1) I think that discovering failures about missing archives seems outside of the scope of performance metrics, I don't have a lot of experience in how people use prometheus though.

Reply:
The case I want to catch is if people are reading from an archive, and it suddenly goes missing or server starts returning 500 errors we want to able to see which archive is no longer working.  We could limit the per-archive metrics to only archives that we've gotten an OK response for the root directory, but that would miss if the server restarts and fails to read at startup... it would also mean 2 metrics you need to alert on to know if ties are down

Reply:
Is there a way to limit the universe of stored tileset names to ~100?

Reply:
sounds good, changed bucket and overall request metrics to set archive to "" on not found errors.